File: lcyl-java/lcyl/lcyl-system/src/main/java/com/lcyl/system/domain/CheckInConfig.java
package com.lcyl.system.domain;


import lombok.Data;

import java.math.BigDecimal;
import java.util.Date;

/**
 * 入住配置表实体
 */
@Data

public class CheckInConfig {

    /**
     * 主键
     */

    private Long id;

    /**
     * 老人ID
     */
    private Long elderId;

    /**
     * 入住开始时间
     */
    private Date checkInStartTime;

    /**
     * 入住结束时间
     */
    private Date checkInEndTime;

    /**
     * 护理等级ID
     */
    private Long nursingLevelId;

    /**
     * 床位号
     */
    private String bedNo;

    /**
     * 费用开始时间
     */
    private Date costStartTime;

    /**
     * 费用结束时间
     */
    private Date costEndTime;

    /**
     * 押金（元）
     */
    private BigDecimal depositAmount;

    /**
     * 护理费用（元/月）
     */
    private BigDecimal nursingCost;

    /**
     * 床位费用（元/月）
     */
    private BigDecimal bedCost;

    /**
     * 其他费用（元/月）
     */
    private BigDecimal otherCost;

    /**
     * 医保支付（元/月）
     */
    private BigDecimal medicalInsurancePayment;

    /**
     * 政府补贴（元/月）
     */
    private BigDecimal governmentSubsidy;

    /**
     * 创建时间
     */
    private Date createTime;

    /**
     * 更新时间
     */
    private Date updateTime;

    /**
     * 创建人id
     */
    private Long createBy;

    /**
     * 更新人id
     */
    private Long updateBy;

    /**
     * 备注
     */
    private String remark;

    /**
     * 床位ID
     */
    private Long bedId;
}

File: lcyl-vue/lcyl/src/api/system/checkInConfig.js
import request from '@/utils/request'

/**
 * 入住配置相关 API（入住配置表 check_in_config）
 */

// 查询入住配置列表
export function listCheckInConfig(query) {
  return request({
    url: '/system/checkin-config/list',
    method: 'get',
    params: query
  })
}

// 查询入住配置详细
export function getCheckInConfig(id) {
  return request({
    url: '/system/checkin-config/' + id,
    method: 'get'
  })
}

// 新增入住配置
export function addCheckInConfig(data) {
  return request({
    url: '/system/checkin-config',
    method: 'post',
    data: data
  })
}

// 修改入住配置
export function updateCheckInConfig(data) {
  return request({
    url: '/system/checkin-config',
    method: 'put',
    data: data
  })
}

// 删除入住配置
export function delCheckInConfig(ids) {
  return request({
    url: '/system/checkin-config/' + ids,
    method: 'delete'
  })
}

/**
 * 提交入住配置：保存到入住配置表，并将老人关联到所选床位（占用床位）
 * @param {Object} data - { elderId, bedId, bedNo, checkInStartTime, checkInEndTime, nursingLevelId, costStartTime, costEndTime, depositAmount, nursingCost, bedCost, otherCost, medicalInsurancePayment, governmentSubsidy }
 */
export function submitCheckInConfig(data) {
  return request({
    url: '/system/checkin-config/submit',
    method: 'post',
    data: data
  })
}
